### ISSUE TITLE
[feature request] point options

### ISSUE THREAD

**Issue by mdsumner** [r-spatial/mapedit#59]
This became a  clear need when I digitized the nodes in this image: 

https://gist.github.com/mdsumner/449148a78ccea02b19ab1a21bfc5acb3

I really need a point that's not a big icon. I ended up digitizing it "top - down" so that each point didn't obscure points to come. :)

(I also forgot to assign the result, so this was a happy and rare use of .Last.value for me!).

**Comment by timelyportfolio** [comment]
I agree completely and will explore.  Thanks for the example.   These are critical to the improvement of mapedit.

**Comment by timelyportfolio** [comment]
The newest version includes circle marker handler.

![image](https://user-images.githubusercontent.com/837910/28758140-b85cb330-7558-11e7-8891-5c4f42afbf34.png)

**Comment by mdsumner** [comment]
Awesome!

**Comment by mdsumner** [comment]
Makes me realize too that topological shared-edit by vertex is easy enough,  but needs internal logic to deduplicate coords. Live feedback requires a lot of object re-creation but moving points alone will be a useful prototype, especially with snap available.

**Comment by timelyportfolio** [comment]
I have tried and unfortunately only come up with these two bad options.  Will need to install from Github (sorry, but found a bug).

### drawMarkerOptions

I would be inclined to add something like this as the default in `mapedit`, but unfortunately the icon is placed just down and to the right of the crosshair.  I tried adjusting all the options and couldn't find anything that would affect the placement.

```
library(mapview)
library(mapedit)
library(leaflet.extras)

map <- mapview()@map %>%
  addDrawToolbar(
    markerOptions = drawMarkerOptions(
      makeIcon(
        iconUrl = "data:;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAABmJLR0QA/wD/AP+gvaeTAAAAiklEQVQ4jc2STQqAIBCFP4KIrlAd0LpVl+l30bZT2BFqoUYLG3UTPXggOO9Tx4E/qgBaYAIO6xFQdk9UA6zA+eIFqKWTpbDzBpQ+QBcRdlY+wJQAGHyAIwGgXSh7a0isnoA9IXfXPgF9AsBbm2OaE3r/jDBQNWZYpEGqQtcrMP88YLqt7TpqlL/XBUh2UGg2SG2jAAAAAElFTkSuQmCC"
      )
    )
  )
  
editMap(map)

### super hack to use L.divIcon

As the title suggests, this one is clearly a hack, but it does work.

```
map <- mapview()@map %>%
  addDrawToolbar(
    markerOptions = drawMarkerOptions(
      makeIcon(
        # just a dummy to construct the arguments
        iconUrl = ""
      )
    )
  )

pos <- Position(function(cl){cl$method=="addDrawToolbar"}, map$x$calls)

map$x$calls[[pos]]$args[[3]]$draw$marker$markerIconFunction = htmlwidgets::JS(
"
function() {return L.divIcon();}
"
)

editMap(map)
```

**Comment by tim-salabim** [comment]
Given that digitizing is nice, but only really useful with snapping, I would suggest not to put too much effort into this. All this will need a major overhaul when leaflet 1.x becomes available. Hence, if that 'hack' can be generalised so that it works with user supplied maps also we could just accept it as the hack it is. If not, then we should probably just stick to what's available for the time being.

**Comment by mdsumner** [comment]
did we have an open issue somewhere for shared-edit / topology-edit?   I've put together a very primitive example: 

https://gist.github.com/mdsumner/4e9ebb4ed9fa03682e58128afa66a16c

I was sure there was a discussio nalready, so I'll expand in there or create a new one if not.

**Comment by edzer** [comment]
Potentially great to have, but depending on packages not on CRAN...

**Comment by mdsumner** [comment]
That is by far the least of the obstacles, the code is easily separable for either mapedit or a kind of API for sf round-trip. I've put some extended notes at that gist and will update as I go. Feedback  is welcome.

**Comment by tim-salabim** [comment]
As from my/our package development perspective, this is something to keep in the back of our heads, but mapedit is far too immature to tackle this at the moment. Let's get the [basics working solidly and flexibly](https://github.com/r-spatial/mapedit/issues/62) before thinking about topological editing. Also, I'd like to wait with this until leaflet has migrated to 1.x as for the moment we can't even snap to vertices...

**Comment by mdsumner** [comment]
Thanks @tim-salabim that is no problem at all. It's already a solution for what I need, just a heads up not a request.
